### ISSUE TITLE
vipsthumbnail has trouble resizing a particular animated gif

### ISSUE THREAD

**Issue by acrispino** [libvips/libvips#61]
Using vips 7.32.3 on ubuntu 13.04

If I try to resize this animated gif: http://i.imgur.com/2CIAU.gif using vipsthumbnail, the output seems to contain all the frames of animation instead of just the first.

I've tried a couple of other animated gifs and they seem to work as I would expect, so I'm not sure if it's a bug in vips or a malformed gif.

**Comment by djiesamsoe** [comment]
it still happens to me, I tried it with a command like this:
```time vipsthumbnail -s 100 gif.gif -p bicubic --vips-concurrency=1 -d -o tre.gif```

What should I do?
is there anything wrong with this command?

Thanks
DjieSamSoe

**Comment by jcupitt** [comment]
Hi, vips uses libmagick to load gif images, so I guess you're using a vips without that compiled in.

What platform are you running on, how did you get the vips you're using?

**Comment by djiesamsoe** [comment]
Hi Jhon,
thanks for fast response.

I'm using Python 2.7, ubuntu 14.10

This case is only an experiment for me, if that is not possible, I do not object to my fund will try to use another library.

Thanks again Jhon.

DjieSamSoe

**Comment by jcupitt** [comment]
Strange, it should work in 14.10, I'll try here in a VM.

**Comment by djiesamsoe** [comment]
yes, there may be a shortage when installing
Thanks for your reply Jhon.

**Comment by jcupitt** [comment]
I've made a 14.10 VM, but there seems to be some kind of DNS problem at the moment and it can't install some packages. I'll try again later.

14.10 is a strange ubuntu to be using, it's no longer supported and is not receiving any security updates. I guess you're certain it's not 14.04?

**Comment by djiesamsoe** [comment]
No problem jhon.
let me try again.
i think this my bad..!

Thanks

**Comment by jcupitt** [comment]
Oh hang on, I just read your command. vips can't write gif files, it can only read them. Try:

```
time vipsthumbnail -s 100 gif.gif -d -o tre.jpg
```

I think I wouldn't use any other options.

**Comment by djiesamsoe** [comment]
if from gif file to jpg file, it works well.
however, I want the end result of a gif file.
gif to gif.

**Comment by jcupitt** [comment]
vipsthumbnail is useful for formats like png and tiff, where things like imagemagick will use very large amounts of memory. For example:

```
$ time vipsthumbnail Chicago.png 
real	0m4.061s
user	0m3.984s
sys	0m0.080s
peak memory: 128MB
$ time convert Chicago.png -resize 128x128 tn_Chicago.jpg
real	0m6.119s
user	0m7.824s
sys	0m0.212s
peak memory: 1.2GB
```

Or:

```
$ time vipsthumbnail wtc.tif
real	0m0.432s
user	0m0.384s
sys	0m0.048s
peak memory: 50MB
$ time convert wtc.tif -resize 128x128 tn_wtc.jpg
real	0m1.901s
user	0m3.328s
sys	0m0.124s
peak memory: 600MB
```

**Comment by jcupitt** [comment]
Sorry, gif write is not possible.

**Comment by djiesamsoe** [comment]
ok, thanks Jhon

**Comment by duhaime** [comment]
@jcupitt in 2025 is it possible to write gifs with vips, or does the library delegate to another library for gif writing? 

I'm seeing the exact behavior OP mentioned for certain gifs, while for others I'm seeing successful resizing, so I'm just trying to sort out what might be going on and whether I need to provide additional config options in our context (rails server with `ruby-vips`)

**Comment by jcupitt** [comment]
Hi @duhaime, please don't add to old issues -- open a new one for a new question. 

If I answer here, no one will ever be able to find it.

**Comment by jcupitt** [comment]
(libvips added proper GIF support back in 2020, so it should work -- please open an issue if you've hit a case that fails)
